src/shared/base-filesystem.c

```diff
@@ -186,6 +205,9 @@ int base_filesystem_create_fd(int fd, const char *root, uid_t uid, gid_t gid) {
                 if (uid_is_valid(uid) || gid_is_valid(gid))
                         if (fchownat(fd, i->dir, uid, gid, AT_SYMLINK_NOFOLLOW) < 0)
                                 return log_error_errno(errno, "Failed to chown %s/%s: %m", root, i->dir);
+
+                if (i->modifier == BASE_FILESYSTEM_EMPTY_MARKER)
+                        empty_fs = true;
```

Comment:
This should be flipped immediately after the faccessat(i>dir), before i->target check. The fs shall be marked as empty even if no target exists.

(I know realistically it doesn't matter much, but feels more correct that way)